crates/miden-tx/src/executor/mod.rs

```diff
@@ -363,32 +395,58 @@ where
         )
         .map_err(TransactionExecutorError::TransactionProgramExecutionFailed);
 
+        let (_, _, _, _, input_notes) = tx_inputs.into_parts();
         match result {
-            Ok(_) => Ok(NoteAccountExecution::Success),
-            Err(tx_execution_error) => {
+            Ok(_) => {
+                // Return all the input notes as successful.
+                Ok(NoteConsumptionInfo::new_successful(
+                    input_notes.into_iter().map(|note| note.into_note()).collect::<Vec<_>>(),
+                ))
+            },
+            Err(error) => {
                 let notes = host.tx_progress().note_execution();
 
-                // empty notes vector means that we didn't process the notes, so an error
-                // occurred somewhere else
+                // Empty notes vector means that we didn't process the notes, so an error
+                // occurred.
                 if notes.is_empty() {
-                    return Err(tx_execution_error);
+                    return Err(error);
                 }
```

Comment:
Do we still need this check here? We're already checking `note.is_empty` at the beginning of the function now.

Reply:
Ah, I think I didn't realize that `notes` was redeclared. Makes sense!

Reply:
Yea was thinking might be a good idea to rename that variable